crates/core/src/views.rs

```diff
@@ -19,35 +22,43 @@ fn powersync_view_sql_impl(
 ) -> Result<String, SQLiteError> {
     let db = ctx.db_handle();
     let table = args[0].text();
-    let statement = extract_table_info(db, table)?;
+    let table_info = TableInfo::parse_from(db, table)?;
 
-    let name = statement.column_text(0)?;
-    let view_name = statement.column_text(1)?;
-    let local_only = statement.column_int(2)? != 0;
+    let name = &table_info.name;
+    let view_name = &table_info.view_name;
+    let local_only = table_info.flags.local_only();
+    let include_metadata = table_info.flags.include_metadata();
 
     let quoted_name = quote_identifier(view_name);
     let internal_name = quote_internal_name(name, local_only);
 
-    let stmt2 = db.prepare_v2("select json_extract(e.value, '$.name') as name, json_extract(e.value, '$.type') as type from json_each(json_extract(?, '$.columns')) e")?;
-    stmt2.bind_text(1, table, sqlite::Destructor::STATIC)?;
+    let mut columns = ColumnNameAndTypeStatement::new(db, table)?;
+    let mut iter = columns.streaming_iter();
 
     let mut column_names_quoted: Vec<String> = alloc::vec![];
     let mut column_values: Vec<String> = alloc::vec![];
     column_names_quoted.push(quote_identifier("id"));
     column_values.push(String::from("id"));
-    while stmt2.step()? == ResultCode::ROW {
-        let name = stmt2.column_text(0)?;
-        let type_name = stmt2.column_text(1)?;
+    while let Some(row) = iter.next() {
+        let ColumnInfo { name, type_name } = row.clone()?;
         column_names_quoted.push(quote_identifier(name));
 
         let foo = format!(
-            "CAST(json_extract(data, {:}) as {:})",
+            "CAST(powersync_extract(data, {:}) as {:})",
```

Comment:
An advantage of the old `json_extract` form was that you could read a powersync db without loading the extension. Is there a nice way to keep that option available, for example when binary data isn't used?